sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -144,6 +223,60 @@ public static IDisposable CreateHttpMessagePropertiesScope(IDictionary<string, o
             return CurrentHttpMessagePropertiesScope.Value;
         }
 
+        private ReadOnlyMemory<HttpPipelinePolicy> CreateRequestPipeline(HttpPipelinePolicy[] policies, List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)> customPolicies)
+        {
+            if (!_internallyConstructed)
+            {
+                throw new InvalidOperationException("Cannot send messages with per-request policies if the pipeline wasn't constructed with HttpPipelineBuilder.");
+            }
+
+            // Copy over client policies and splice in custom policies at designated indices
+            int transportIndex = _pipeline.Length - 1;
+            var pipeline = _pipeline.Span;
```

Comment:
Could this line be above the previous one and then the previous line would be able to use the local?